src/Common/tests/Tests/System/StringTests.cs

```diff
@@ -1315,7 +1316,293 @@ public static void ContainsUnknownComparisonType_StringComparison()
+        public static void SequenceCompareToArrayImplicit_Char()
+        {
+            string s1 = "456";
+            int result = s1.CompareTo(s1);
+            Assert.Equal(0, result);
+
+            ReadOnlySpan<char> first = s1.AsSpan(0, 3);            
+            result = first.SequenceCompareTo<char>(s1.AsSpan());
+            Assert.Equal(0, result);
+        }
+
+        [Fact]
+        public static void SequenceCompareToArraySegmentImplicit_Char()
+        {
+            char[] src = { '1', '2', '3' };
+            char[] dst = { '5', '1', '2', '3', '9' };
+            var segment = new ArraySegment<char>(dst, 1, 3);
+
+            string s1 = new string(src, 0, 3);
+            int result = s1.CompareTo(segment.ToString());
+
+            ReadOnlySpan<char> first = s1.AsSpan();
+            result = first.SequenceCompareTo<char>(segment);
+            Assert.Equal(0, result);
+        }
+
+        [Fact]
+        public static void LengthMismatchSequenceCompareTo_Char()
+        {
+            string value = "456";
+
+            string s1 = value.Substring(0, 2);
+            string s2 = value.Substring(0, 3);
+            int result = s1.CompareTo(s2);
+            Assert.True(result < 0);
+
+            result = s2.CompareTo(s1);
+            Assert.True(result > 0);
+
+            // one sequence is empty
+            s1 = value.Substring(1, 0);
+
+            result = s1.CompareTo(s2);
+            Assert.True(result < 0);
+
+            result = s2.CompareTo(s1);
+            Assert.True(result > 0);
+
+            ReadOnlySpan<char> first = s1.AsSpan();
+            ReadOnlySpan<char> second = s2.AsSpan();
+            result = first.SequenceCompareTo<char>(second);
+            Assert.True(result < 0);
+
+            result = second.SequenceCompareTo<char>(first);
+            Assert.True(result > 0);
+
+            // one sequence is empty
+            first = value.AsSpan(1, 0);
+
+            result = first.SequenceCompareTo<char>(second);
+            Assert.True(result < 0);
+
+            result = second.SequenceCompareTo<char>(first);
+            Assert.True(result > 0);
+        }
+
+        [Fact]        
+        public static void SequenceCompareToWithSingleMismatch_Char()
+        {
+            for (int length = 1; length < 32; length++)
+            {
+                for (int mismatchIndex = 0; mismatchIndex < length; mismatchIndex++)
+                {
+                    var first = new char[length];
+                    var second = new char[length];
+                    for (int i = 0; i < length; i++)
+                    {
+                        first[i] = second[i] = (char)(i + 1);
+                    }
+
+                    second[mismatchIndex] = (char)(second[mismatchIndex] + 1);
+
+                    string s1 = new string(first);
+                    string s2 = new string(second);
+                    int result = string.Compare(s1, s2, StringComparison.Ordinal);                    
```

Comment:
NB. We need to use `string.Compare` because `s1.CompareTo` is culture sensitive.